### PR TITLE
PDPDEVTOOL-4977: Add suitescript 2.x unit test stubs 1

### PR DESCRIPTION
**Pull request by lfpvillegas**

### CODE REVIEW

File: packages/unit-testing/stubs/workbook/workbook.js [hunk 1]
```diff
@@ -0,0 +1,927 @@
...
+	'./Section',
+	'./Series',
+	'./Sort',
+	'./SortByDataDimensionItem',
+	'./SortDefinition',
+	'./TableColumn',
+	'./TableColumnCondition',
+	'./TableColumnFilter',
+	'./WorkbookInstance',
+], function () {
```
Comment (davidecorreu) [inline packages/unit-testing/stubs/workbook/workbook.js:45]: All the elements defined above should be parameters of this function. They also need to be defined in the SuiteCloudJestConfiguration.js as modules.

File: packages/unit-testing/stubs/log/logGlobalObject.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+/**
```
Comment (davidecorreu) [inline packages/unit-testing/stubs/log/logGlobalObject.js:1]: I don't know where this one comes from, I don't see it on the documentation. Also is not following AMD structure that starts with `define([...`

Comment (lfpvillegas) [reply packages/unit-testing/stubs/log/logGlobalObject.js:1]: This was exactly the omment I put in slack. 

`"Help mentions that log is both a global object but for testing purposes you could use N/log module. I have created the N/log module, but in the Suitescript 2.0 original library it appeared log as object. Do we use that instead?"`

File: packages/unit-testing/jest-configuration/SuiteCloudJestConfiguration.js [hunk 3]
```diff
@@ -181,6 +189,10 @@ const CORE_STUBS = [
 		module: 'N/https',
 		path: `<rootDir>/node_modules/${CORE_STUBS_PATH}/https/https.js`,
 	},
+	{
+		module: 'N/https/clientCertificate',
+		path: `<rootDir>/node_modules/${CORE_STUBS_PATH}/https/clientCertificate.js`,
+	},
```
Comment (davidecorreu) [inline packages/unit-testing/jest-configuration/SuiteCloudJestConfiguration.js:195]: Remove this one, we have moved it into `/https/clientCertificate/clientCertificate.js` and it's arleady defined at lines 208-211.

File: packages/unit-testing/jest-configuration/SuiteCloudJestConfiguration.js [hunk 4]
```diff
@@ -357,6 +377,210 @@ const CORE_STUBS = [
...
+	{
+		module: 'N/workbook/tableColumnCondition',
+		path: `<rootDir>/node_modules/${CORE_STUBS_PATH}/workbook/TableColumnCondition.js`,
+	},
+	{
+		module: 'N/workbook/tableColumnFilter',
+		path: `<rootDir>/node_modules/${CORE_STUBS_PATH}/workbook/TableColumnFilter.js`,
+	},
+	{
+		module: 'N/workbook/workbook',
```
Comment (davidecorreu) [inline packages/unit-testing/jest-configuration/SuiteCloudJestConfiguration.js:581]: The module here should be `N/workbook/instance`.

File: packages/unit-testing/stubs/workbook/Aspect.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+define(['./CalculatedMeasure', './DataMeasure'], function (CalculatedMeasure, DataMeasure) {
```
Comment (davidecorreu) [inline packages/unit-testing/stubs/workbook/Aspect.js:1]: This workbook "subObject" is not appearing in documentation. I think we should leave it as it is, but leaving the comment to keep track of it.

File: packages/unit-testing/stubs/workbook/Expression.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
+define([], function () {
+	/**
+	 * @class Expresion
+	 * @classDescription An expression. An expression can be used when you create a pivot definition, a data dimension item, a measure, a conditional filter, or a constant.
+	 * @constructor
+	 * @protected
+	 *
+	 * @since 2021.2
+	 */
+	function Expresion() {
```
Comment (Extend-Apps) [inline packages/unit-testing/stubs/workbook/Expression.js:10]: Misspelling... should be **Expression**

File: packages/unit-testing/stubs/workbook/Chart.js [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+) {
+	/**
+	 * @class Chart
+	 * @classDescription Object representing a chart of a workbook
+	 * @constructor
+	 * @protected
+	 *
+	 * @since 2020.2
+	 */
+	function Chart() {
```
Comment (davidecorreu) [inline packages/unit-testing/stubs/workbook/Chart.js:18]: Not in docs. No action needed.

File: packages/unit-testing/stubs/workbook/CalculatedMeasure.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
+define(['./Expression'], function () {
```
Comment (davidecorreu) [inline packages/unit-testing/stubs/workbook/CalculatedMeasure.js:1]: Missing Expression param in function
`define(['./Expression'], function (Expression) {`

File: packages/unit-testing/stubs/workbook/ChartAxis.js [hunk 9]
```diff
@@ -0,0 +1,39 @@
+define(['./Expression'], function () {
+	/**
+	 * @class ChartAxis
+	 * @classDescription Axis of a chart
+	 * @constructor
+	 * @protected
+	 *
+	 * @since 2020.2
+	 */
+	function ChartAxis() {
```
Comment (davidecorreu) [inline packages/unit-testing/stubs/workbook/ChartAxis.js:10]: Not in docs. No action needed.

File: packages/unit-testing/stubs/workbook/ChartAxis.js [hunk 10]
```diff
@@ -0,0 +1,39 @@
+define(['./Expression'], function () {
```
Comment (davidecorreu) [inline packages/unit-testing/stubs/workbook/ChartAxis.js:1]: Missing Expression param in function.

File: packages/unit-testing/stubs/workbook/Expression.js (hunk 6 above)
Comment (vic-nik) [reply packages/unit-testing/stubs/workbook/Expression.js:10]: Corrected. Thank you Darren! @Extend-Apps

File: packages/unit-testing/stubs/workbook/Category.js [hunk 11]
```diff
@@ -0,0 +1,57 @@
+define(['./ChartAxis', './Section', './DataDimension', './SortDefinition'], function (ChartAxis, Section, DataDimension, SortDefinition) {
+	/**
+	 * @class Category
+	 * @classDescription Category of a chart
+	 * @constructor
+	 * @protected
+	 *
+	 * @since 2020.2
+	 */
+	function Category() {
```
[the remark this answers is not in the document]
Comment (vic-nik) [reply packages/unit-testing/stubs/workbook/Category.js:10]: The generated JSON file has createCategory method on workbook module that returns this object

[the remark this answers is not in the document]
Comment (vic-nik) [reply packages/unit-testing/stubs/workbook/Category.js:10]: There's more references of it as well (not in the help though)

File: packages/unit-testing/stubs/workbook/CalculatedMeasure.js (hunk 8 above)
Comment (vic-nik) [reply packages/unit-testing/stubs/workbook/CalculatedMeasure.js:1]: Corrected

File: packages/unit-testing/stubs/workbook/Chart.js (hunk 7 above)
Comment (vic-nik) [reply packages/unit-testing/stubs/workbook/Chart.js:18]: Same as with Category, e.g. it's used as return type of createChart function of the N/workbook

File: packages/unit-testing/stubs/workbook/ChartAxis.js (hunk 10 above)
Comment (vic-nik) [reply packages/unit-testing/stubs/workbook/ChartAxis.js:1]: Corrected

File: packages/unit-testing/stubs/workbook/Aspect.js (hunk 5 above)
Comment (vic-nik) [reply packages/unit-testing/stubs/workbook/Aspect.js:1]: Same as with Category and Chart, there's createAspect function in N/workbook

File: packages/unit-testing/stubs/workbook/Pivot.js [hunk 12]
```diff
@@ -0,0 +1,127 @@
+define(['../dataset/Dataset', '../datasetLink/DatasetLink', './Expression', './LimitingFilter', './ConditionalFilter', './PivotAxis', './ReportStyle'], function (
+	Dataset,
+	DatasetLink,
```
Comment (vic-nik) [inline packages/unit-testing/stubs/workbook/Pivot.js:3]: Expression param is missing as function argument

File: packages/unit-testing/stubs/workbook/PivotResultsIterator.js [hunk 13]
```diff
@@ -0,0 +1,12 @@
+define([], function () {
```
Comment (vic-nik) [inline packages/unit-testing/stubs/workbook/PivotResultsIterator.js:1]: This module is not used anywhere. It is probably the return type of the loadPivotResults function of the N/workbook but it's not documented. Let's keep it and improve later if needed.

File: packages/unit-testing/stubs/workbook/Table.js [hunk 14]
```diff
@@ -0,0 +1,58 @@
+define(['./Dataset', './Expression', './TableColumn'], function (Dataset, TableColumn) {
```
Comment (vic-nik) [inline packages/unit-testing/stubs/workbook/Table.js:1]: Expression is missing as function arg

File: packages/unit-testing/stubs/dataset/Column.js [hunk 15]
```diff
@@ -1,4 +1,4 @@
-define([], function () {
+define(['../workbook/Expression'], function () {
```
Comment (davidecorreu) [inline packages/unit-testing/stubs/dataset/Column.js:1]: Missing Expession parameter in function

File: packages/unit-testing/stubs/dataset/DatasetInstance.js [hunk 16]
```diff
@@ -1,4 +1,4 @@
-define(['./Condition', './Column'], function (Condition, Column) {
```
Comment (davidecorreu) [inline packages/unit-testing/stubs/dataset/DatasetInstance.js:1]: Missing Expression parameter in function.

Comment (davidecorreu) [reply packages/unit-testing/stubs/dataset/DatasetInstance.js:1]: Taking a closer look, I think we should also "inject" query.ResultSet and query.PagedData.

File: packages/unit-testing/stubs/dataset/dataset.js [hunk 17]
```diff
@@ -1,4 +1,4 @@
-define(['./DatasetInstance', './Join', './Column', './Condition'], function (Dataset, Join, Column, Condition) {
+define(['../workbook/Expression', './DatasetInstance', './Join', './Column', './Condition'], function (Dataset, Join, Column, Condition) {
```
Comment (davidecorreu) [inline packages/unit-testing/stubs/dataset/dataset.js:1]: Missing Expression parameter in function

File: packages/unit-testing/stubs/datasetLink/DatasetLinkInstance.js [hunk 18]
```diff
@@ -1,4 +1,4 @@
-define([], function () {
+define(['../workbook/Expression'], function (Expression) {
```
Comment (davidecorreu) [inline packages/unit-testing/stubs/datasetLink/DatasetLinkInstance.js:1]: dataset.DatasetInstance should be "injected" as Dataset parameter.

File: packages/unit-testing/stubs/datasetLink/datasetLink.js [hunk 19]
```diff
@@ -1,4 +1,4 @@
-define(['./DatasetLinkInstance'], function (DatasetLink) {
+define(['../workbook/Expression', './DatasetLinkInstance'], function (DatasetLink) {
```
Comment (davidecorreu) [inline packages/unit-testing/stubs/datasetLink/datasetLink.js:1]: Missing Expression parameter. dataset.DatasetInstance should be also injected as Dataset parameter.

File: packages/unit-testing/stubs/workbook/Chart.js [hunk 20]
```diff
@@ -0,0 +1,163 @@
+define(['../dataset/Dataset', '../datasetLink/DatasetLink', './LimitingFilter', './ConditionalFilter', './Expression', './Legend', './Category', './Series'], function (
```
Comment (davidecorreu) [inline packages/unit-testing/stubs/workbook/Chart.js:1]: Expression paramater missing in function.

File: packages/unit-testing/stubs/workbook/ConditionalFilter.js [hunk 21]
```diff
@@ -0,0 +1,81 @@
+define(['DataMeasure', './CalculatedMeasure', './ChildNodesSelector', './DescendantOrSelfNodesSelector', './Expression', './PathSelector'], function (
+	DataMeasure,
+	CalculatedMeasure,
+	DescendantOrSelfNodesSelector,
+	PathSelector,
+	ChildNodesSelector
+) {
```
Comment (davidecorreu) [inline packages/unit-testing/stubs/workbook/ConditionalFilter.js:7]: Missing `./` in DataMeasure "injection".
Missing Expression parameter in function.
Missing DimensionSelector "injection".
Wrong order position for ChildNodesSelector.
"Injections" and function parameters should be ordered alphabetiacally.

File: packages/unit-testing/stubs/workbook/DataDimension.js [hunk 22]
```diff
@@ -0,0 +1,58 @@
+define(['./DataMeasure', './DataDimensionItem', './CalculatedMeasure'], function (DataMeasure, DataDimensionItem, CalculatedMeasure) {
```
Comment (davidecorreu) [inline packages/unit-testing/stubs/workbook/DataDimension.js:1]: Not following alphabetical order. Should we reorder?

File: packages/unit-testing/stubs/workbook/DataDimensionItem.js [hunk 23]
```diff
@@ -0,0 +1,48 @@
+define(['./Expression',], function () {
```
Comment (davidecorreu) [inline packages/unit-testing/stubs/workbook/DataDimensionItem.js:1]: Missing Expression paramater in function.

File: packages/unit-testing/stubs/workbook/DataDimensionItemValue.js [hunk 24]
```diff
@@ -0,0 +1,48 @@
...
+		 * @type {DataDimension}
+		 * @throws {SuiteScriptError} READ_ONLY_PROPERTY when setting the property is attempted
+		 *
+		 * @since 2021.2
+		 */
+		this.item = undefined;
+		/**
+		 * Value of the data dimension item
+		 * @name DataDimensionItemValue#value
+		 * @type {string|number|boolean|Record|Currency|Range}
```
Comment (davidecorreu) [inline packages/unit-testing/stubs/workbook/DataDimensionItemValue.js:23]: In helps also appears workbook.Durantion a possible return type. Let's leave it as it is. No action needed.

File: packages/unit-testing/stubs/workbook/DataDimensionValue.js [hunk 25]
```diff
@@ -0,0 +1,48 @@
+define(['DataDimension', './DataDimensionItemValue'], function (DataDimension, DataDimensionItemValue) {
```
Comment (davidecorreu) [inline packages/unit-testing/stubs/workbook/DataDimensionValue.js:1]: Missing `./` at beginning of DataDimension "injection".

File: packages/unit-testing/stubs/workbook/DataMeasure.js [hunk 26]
```diff
@@ -0,0 +1,74 @@
+define(['./Expression',], function () {
```
Comment (davidecorreu) [inline packages/unit-testing/stubs/workbook/DataMeasure.js:1]: Missing Expression parameter in function.

File: packages/unit-testing/stubs/workbook/DimensionSelector.js [hunk 27]
```diff
@@ -0,0 +1,39 @@
+define(['Section', './DataDimension'], function (Section, DataDimension) {
```
Comment (davidecorreu) [inline packages/unit-testing/stubs/workbook/DimensionSelector.js:1]: Not in alphabetiaclly order.

File: packages/unit-testing/stubs/workbook/Duration.js [hunk 28]
```diff
@@ -0,0 +1,48 @@
+define([], function () {
```
Comment (davidecorreu) [inline packages/unit-testing/stubs/workbook/Duration.js:1]: No action needed.
Missing DimensionSort Objet before this one according to helpdocs.

File: packages/unit-testing/stubs/dataset/dataset.js (hunk 17 above)
Comment (davidecorreu) [reply packages/unit-testing/stubs/dataset/dataset.js:1]: Also alphabetical order is not respected.

File: packages/unit-testing/stubs/workbook/ConditionalFilter.js (hunk 21 above)
Comment (davidecorreu) [reply packages/unit-testing/stubs/workbook/ConditionalFilter.js]: DimensionSelector still missing.

File: packages/unit-testing/stubs/workbook/ConditionalFilter.js [hunk 29]
```diff
@@ -0,0 +1,82 @@
+define(['./DataMeasure', './CalculatedMeasure', './ChildNodesSelector', './DescendantOrSelfNodesSelector', './Expression', './PathSelector'], function (
```
Comment (davidecorreu) [inline packages/unit-testing/stubs/workbook/ConditionalFilter.js:1]: Missing DimensionSelector "injection"

**Review by vic-nik — APPROVED** [review]

**Review by davidecorreu — APPROVED** [review]

**Pull request merged**